src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Exceptions/FhirServiceExceptionProcessor.cs

```diff
@@ -0,0 +1,82 @@
+
+namespace Microsoft.Health.Extensions.Fhir.Telemetry.Exceptions
+{
+    public static class FhirServiceExceptionProcessor
+    {
+        private static readonly IExceptionTelemetryProcessor _exceptionTelemetryProcessor = new ExceptionTelemetryProcessor();
+
+        public static void ProcessException(Exception exception, ITelemetryLogger logger)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            var (customException, errorName) = CustomizeException(exception);
+
+            logger.LogError(customException);
+
+            string exceptionName = customException.Equals(exception) ? $"{ErrorType.FHIRServiceError}{errorName}" : customException.GetType().Name;
+            _exceptionTelemetryProcessor.LogExceptionMetric(customException, logger, FhirClientMetrics.HandledException(exceptionName, ErrorSeverity.Critical));
+        }
+
+        public static (Exception customException, string errorName) CustomizeException(Exception exception)
+        {
+            EnsureArg.IsNotNull(exception, nameof(exception));
+
+            string message;
+            string errorName;
+
+            switch (exception)
+            {
+                case FhirOperationException _:
+                    var status = ((FhirOperationException)exception).Status;
+                    switch (status)
+                    {
+                        case HttpStatusCode.Forbidden:
+                            message = FhirResources.FhirServiceAccessForbidden;
+                            string helpLink = "https://docs.microsoft.com/azure/healthcare-apis/iot/deploy-iot-connector-in-azure#accessing-the-iot-connector-from-the-fhir-service";
+                            errorName = nameof(FhirServiceErrorCode.AuthorizationError);
+                            return (new UnauthorizedAccessFhirServiceException(message, exception, helpLink, errorName), errorName);
+                        case HttpStatusCode.NotFound:
+                            message = FhirResources.FhirServiceNotFound;
+                            errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                            return (new InvalidFhirServiceException(message, exception, errorName), errorName);
+                        default:
+                            return (exception, status.ToString());
+                    }
+
+                case UriFormatException _:
+                    message = FhirResources.FhirServiceUriFormatInvalid;
+                    errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                    return (new InvalidFhirServiceException(message, exception, errorName), errorName);
+
+                case HttpRequestException _:
+                    message = FhirResources.FhirServiceHttpRequestError;
+                    errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                    return (new InvalidFhirServiceException(message, exception, errorName), errorName);
```

Comment:
Should we look for a specific error code in HttpRequestException? I wonder if it is too broad to assume that it is a config error, as even intermittent 503s etc could throw this?

Reply:
`HttpRequestException`'s error code property, `StatusCode`, applies to only .NET 5 and later.

In commit 9ab3837, I've scoped `HttpRequestException` down by checking the `Message` property for now and added a TODO to check for `StatusCode` in .NET 5 and later.